source/vision/__init__.py

```diff
@@ -0,0 +1,77 @@
+#vision/__init__.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2018-2019 NV Access Limited, Babbage B.V.
+
+"""Framework to facilitate changes in how content is displayed on screen.
+One or more assistant functions can be implemented in vision enhancement providers.
+Add-ons can provide their own provider
+using modules in the visionEnhancementProviders package containing a L{VisionEnhancementProvider} class.
+"""
+
+from .constants import *
+from .providerBase import VisionEnhancementProvider
+from .visionHandler import VisionHandler, getProviderClass
+import pkgutil
+import visionEnhancementProviders
+import config
+from logHandler import log
+
+def initialize():
+	global handler
+	config.addConfigDirsToPythonPackagePath(visionEnhancementProviders)
+	handler = VisionHandler()
+
+def pumpAll():
+	"""Runs tasks at the end of each core cycle."""
+	if handler and handler.extensionPoints:
+		handler.extensionPoints.post_coreCycle.notify()
+
+def terminate():
+	global handler
+	handler.terminate()
+	handler = None
+
+def getProviderList(onlyStartable=True):
+	"""Gets a list of available vision enhancement names with their descriptions as well as supported roles.
+	@param onlyStartable: excludes all providers for which the check method returns C{False}.
+	@type onlyStartable: bool
+	@return: list of tuples with provider names, provider descriptions, and supported roles.
+		See L{constants.Role} for the available constants.
+	@rtype: [(str,unicode,str])]
+	"""
+	providerList = []
+	for loader, name, isPkg in pkgutil.iter_modules(visionEnhancementProviders.__path__):
+		if name.startswith('_'):
+			continue
+		try:
+			provider = getProviderClass(name)
+		except:
+			# Purposely catch everything.
+			# A provider can raise whatever exception it likes,
+			# therefore it is unknown what to expect.
+			log.error(
+				"Error while importing vision enhancement provider %s" % name,
+				exc_info=True
+			)
+			continue
+		try:
+			if not onlyStartable or provider.check():

```

Comment:
You had a comment about using the word check as a name for the function. I'm very reluctant to change this as in fact, check is a method on driverHandler.Driver. In other words, check is used this way for BrailleDisplayDrivers and SynthDrivers as well. I think I prefer consistency over clarity here.

Reply:
Consistently unclear? I missed that it's on the `Driver` base class.

This is an opportunity to adapt the interface, and provide further clarity. Consider the following psudo-code approach:

``` Python
class VisionEnhancementProvider(driverHandler.driver):

	@abstractclassmethod
	def canStart():
		raise NotImplementedError

	def check():
		return self.canStart()

```
You might even want to re-map `check` to `canStart` in `__new__` or `__init__` if you wanted to save the extra function call overhead, but I doubt the performance cost justifies the cost to readability.

Reply:
Unfortunately, there is no such thing as an abstract class method.

HOw about implementing canStart for braille and speech as well?

Then:

* We rename check to canStart
* Calling check  raises a deprecation warning
* The base method of canStart calls check on classes where check is implemented/available.

Reply:
May be isAvailable is a better name then canStart? see the doc string of check on driverHandler.Driver

Reply:
A [stackoverflow answer](https://stackoverflow.com/a/11218474/346188) presents a solution for an abstract class method. This won't stop you from calling 
`VisionEnhancementProvider.canStart()`, you need `raise NotImplementedError` for that, but it will stop you from creating an instance of the class that does not override the class method. It also communicates the intent quite well.

> HOw about implementing canStart for braille and speech as well?

I think I would prefer not to extend the scope of the PR that far, it would logically follow from there that we get rid of `check` entirely. 

> May be isAvailable is a better name then canStart? see the doc string of check on driverHandler.Driver

Perhaps, but only if this really fits what we want the providers to tell us, this method should be adapting the generic `Driver.check` function, to make more sense in the context of a `VisionEnhancementProvider`. I think we want them to tell us more than "are they available", but also if they can actually be started / run. Say two providers require the same resources, they can't both run at once, this is their chance to say they are not able to. Am I understanding the design correctly?